src/mps/dmrg.jl

```diff
@@ -278,7 +278,8 @@ function dmrg(PH, psi0::MPS, sweeps::Sweeps; kwargs...)
         end
 
         energy = vals[1]
-        phi::ITensor = vecs[1]
+        ## Right now this convert problem is prevelent in an eigen solver implemented in KrylovKit
+        phi::ITensor = itensor(convert(typeof(tensor(phi)), tensor(vecs[1])))
```

Comment:
I don't understand this comment, and also I don't understand what issue this code is fixing.